src/main/java/com/zendesk/maxwell/bootstrap/SynchronousBootstrapper.java

```diff
@@ -0,0 +1,221 @@
+package com.zendesk.maxwell.bootstrap;
+
+import com.zendesk.maxwell.BinlogPosition;
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.MaxwellReplicator;
+import com.zendesk.maxwell.RowMap;
+import com.zendesk.maxwell.producer.AbstractProducer;
+import com.zendesk.maxwell.schema.Database;
+import com.zendesk.maxwell.schema.Schema;
+import com.zendesk.maxwell.schema.Table;
+import com.zendesk.maxwell.schema.columndef.ColumnDef;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class SynchronousBootstrapper extends AbstractBootstrapper {
+
+	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellReplicator.class);
+	private static final long INSERTED_ROWS_UPDATE_PERIOD_MILLIS = 250;
+
+	private long lastInsertedRowsUpdateTimeMillis = 0;
+
+	public SynchronousBootstrapper(MaxwellContext context) { super(context); }
+
+	@Override
+	public boolean shouldSkip(RowMap row) {
+		// the synchronous bootstrapper blocks other incoming messages
+		// to the replication stream so there's nothing to skip
+		return false;
+	}
+
+	@Override
+	public void startBootstrap(RowMap startBootstrapRow, AbstractProducer producer, MaxwellReplicator replicator) throws Exception {
+		String databaseName = ( String ) startBootstrapRow.getData("database_name");
+		String tableName = ( String ) startBootstrapRow.getData("table_name");
+		LOGGER.debug(String.format("bootstrapping request for %s.%s", databaseName, tableName));
+		Schema schema = replicator.getSchema();
+		Database database = findDatabase(schema, databaseName);
+		Table table = findTable(tableName, database);
+		BinlogPosition position = startBootstrapRow.getPosition();
+		producer.push(startBootstrapRow);
+		producer.push(bootstrapStartRowMap(table, position));
+		LOGGER.info(String.format("bootstrapping started for %s.%s, binlog position is %s", databaseName, tableName, position.toString()));
+		try ( Connection connection = getConnection() ) {
+			setBootstrapRowToStarted(startBootstrapRow, connection);
+			ResultSet resultSet = getAllRows(databaseName, tableName, schema, connection);
+			int insertedRows = 0;
+			while ( resultSet.next() ) {
+				RowMap row = new RowMap(
+						"insert",
+						databaseName,
+						tableName,
+						System.currentTimeMillis(),
```

Comment:
hey @nmaquet, I realize post-merge that the `ts` row is inconsistent -- the bootstrapped rows use milliseconds, whereas the replicated rows use seconds.  can you fix to use seconds in a separate PR?